### ISSUE TITLE
Values are not obscured as expected

### ISSUE THREAD

**Issue by EdwardMarsDS** [ydataai/pandas-profiling#1165]
### Current Behaviour

When using the sensitive=True flag, data is obscured from the columns in the report, however names appear in category frequency plot.  For the example I've included I've replaced the names with numeric values, however when I first ran the report names were visible in the chart legend. 

![Sensitive Leak](https://user-images.githubusercontent.com/102184868/203427067-e3019748-781f-4e9c-a9a1-31a3d7b6f77a.jpg)

### Expected Behaviour

The legend labels should be either excluded when sensitive = True or the text replaced or obscured as it is in the value table.

### Data Description

This is repeatable with any categorical data

### Code that reproduces the bug

```Python
`profile_df = ProfileReport(final_df, title="Data Profiling Report - Special Test", 
                           samples=None,
                           explorative=False, 
                           correlations={"auto": {"calculate": False},
                                         "pearson": {"calculate": True},
                                         "spearman": {"calculate": False},
                                         "kendall": {"calculate": False},
                                         "phi_k": {"calculate": True},
                                         "cramers": {"calculate": True}
                                        },


                           progress_bar = True,
                           missing_diagrams={"heatmap": True,
                                             "dendrogram": False,
                                             "Count": False},
                           sensitive=True,
                           vars={"num": {"low_categorical_threshold": 5}
                               }
                           )`
```

### pandas-profiling version

pandas-profiling v3.3.0

### Dependencies

```Text
# Name                    Version                   Build  Channel
ansi2html                 1.8.0           py310h5588dad_0    conda-forge
anyio                     3.5.0           py310haa95532_0
argon2-cffi               21.3.0             pyhd3eb1b0_0
argon2-cffi-bindings      21.2.0          py310h2bbff1b_0
asn1crypto                1.5.1                    pypi_0    pypi
asttokens                 2.0.5              pyhd3eb1b0_0    anaconda
attrs                     21.4.0             pyhd3eb1b0_0
babel                     2.9.1              pyhd3eb1b0_0
backcall                  0.2.0              pyhd3eb1b0_0    anaconda
beautifulsoup4            4.11.1          py310haa95532_0
bleach                    4.1.0              pyhd3eb1b0_0
brotli-python             1.0.9           py310h8a704f9_7    conda-forge
brotlipy                  0.7.0           py310h2bbff1b_1002
bzip2                     1.0.8                he774522_0    anaconda
ca-certificates           2022.9.24            h5b45459_0    conda-forge
certifi                   2022.9.24          pyhd8ed1ab_0    conda-forge
cffi                      1.15.1          py310h2bbff1b_0
charset-normalizer        2.0.4              pyhd3eb1b0_0
click                     8.1.3           py310h5588dad_0    conda-forge
colorama                  0.4.4              pyhd3eb1b0_0    anaconda
cryptography              36.0.2                   pypi_0    pypi
cycler                    0.11.0                   pypi_0    pypi
dash                      0.39.0                     py_0    plotly
dash-core-components      0.44.0                     py_0    plotly
dash-html-components      0.14.0                     py_0    plotly
dash-renderer             0.20.0                     py_0    plotly
dash-table                3.6.0                      py_0    plotly
debugpy                   1.5.1           py310hd77b12b_0    anaconda
decorator                 5.1.1              pyhd3eb1b0_0    anaconda
defusedxml                0.7.1              pyhd3eb1b0_0
entrypoints               0.4             py310haa95532_0    anaconda
executing                 0.8.3              pyhd3eb1b0_0    anaconda
filelock                  3.8.0                    pypi_0    pypi
flask                     2.2.2              pyhd8ed1ab_0    conda-forge
flask-compress            1.13               pyhd8ed1ab_0    conda-forge
fonttools                 4.37.1                   pypi_0    pypi
htmlmin                   0.1.12                   pypi_0    pypi
idna                      3.3                pyhd3eb1b0_0
imagehash                 4.3.0                    pypi_0    pypi
importlib-metadata        4.11.4          py310h5588dad_0    conda-forge
ipykernel                 6.9.1           py310haa95532_0    anaconda
ipympl                    0.9.2                    pypi_0    pypi
ipython                   8.3.0           py310haa95532_0    anaconda
ipython_genutils          0.2.0              pyhd3eb1b0_1
ipywidgets                8.0.2                    pypi_0    pypi
itsdangerous              2.1.2              pyhd8ed1ab_0    conda-forge
jedi                      0.18.1          py310haa95532_1    anaconda
jinja2                    3.0.3              pyhd3eb1b0_0
joblib                    1.1.0                    pypi_0    pypi
json5                     0.9.6              pyhd3eb1b0_0
jsonschema                4.4.0           py310haa95532_0
jupyter-dash              0.4.2              pyhd8ed1ab_1    conda-forge
jupyter_client            7.2.2           py310haa95532_0    anaconda
jupyter_core              4.10.0          py310haa95532_0    anaconda
jupyter_server            1.18.1          py310haa95532_0
jupyterlab                3.4.4           py310haa95532_0
jupyterlab-widgets        3.0.3                    pypi_0    pypi
jupyterlab_pygments       0.1.2                      py_0
jupyterlab_server         2.12.0          py310haa95532_0
jupyterlab_widgets        1.0.0              pyhd3eb1b0_1
kiwisolver                1.4.4                    pypi_0    pypi
libffi                    3.4.2                hd77b12b_4    anaconda
markupsafe                2.1.1           py310h2bbff1b_0
matplotlib                3.5.3                    pypi_0    pypi
matplotlib-inline         0.1.2              pyhd3eb1b0_2    anaconda
missingno                 0.5.1                    pypi_0    pypi
mistune                   0.8.4           py310h2bbff1b_1000
multimethod               1.8                      pypi_0    pypi
nbclassic                 0.3.5              pyhd3eb1b0_0
nbclient                  0.5.13          py310haa95532_0
nbconvert                 6.4.4           py310haa95532_0
nbformat                  5.3.0           py310haa95532_0
nest-asyncio              1.5.5           py310haa95532_0    anaconda
networkx                  2.8.6                    pypi_0    pypi
nodejs                    18.10.0              h57928b3_0    conda-forge
notebook                  6.4.12          py310haa95532_0
numpy                     1.23.2                   pypi_0    pypi
openssl                   1.1.1q               h8ffe710_0    conda-forge
oscrypto                  1.3.0                    pypi_0    pypi
packaging                 21.3               pyhd3eb1b0_0
pandas                    1.4.4                    pypi_0    pypi
pandas-profiling          3.3.0                    pypi_0    pypi
pandocfilters             1.5.0              pyhd3eb1b0_0
parso                     0.8.3              pyhd3eb1b0_0    anaconda
patsy                     0.5.2                    pypi_0    pypi
phik                      0.12.2                   pypi_0    pypi
pickleshare               0.7.5           pyhd3eb1b0_1003    anaconda
pillow                    9.2.0                    pypi_0    pypi
pip                       21.2.4          py310haa95532_0    anaconda
plotly                    5.10.0                   pypi_0    pypi
powershell_shortcut       0.0.1                         3
prometheus_client         0.14.1          py310haa95532_0
prompt-toolkit            3.0.20             pyhd3eb1b0_0    anaconda
pure_eval                 0.2.2              pyhd3eb1b0_0    anaconda
pyarrow                   8.0.0                    pypi_0    pypi
pycparser                 2.21               pyhd3eb1b0_0
pycryptodomex             3.15.0                   pypi_0    pypi
pydantic                  1.9.2                    pypi_0    pypi
pygments                  2.11.2             pyhd3eb1b0_0    anaconda
pyjwt                     2.4.0                    pypi_0    pypi
pyopenssl                 22.0.0             pyhd3eb1b0_0
pyparsing                 3.0.9           py310haa95532_0
pyrsistent                0.18.0          py310h2bbff1b_0
pysocks                   1.7.1           py310haa95532_0
python                    3.10.4               hbb2ffb3_0    anaconda
python-dateutil           2.8.2              pyhd3eb1b0_0    anaconda
python-dotenv             0.21.0                   pypi_0    pypi
python-fastjsonschema     2.16.2          py310haa95532_0
python_abi                3.10                    2_cp310    conda-forge
pytz                      2022.1          py310haa95532_0
pywavelets                1.3.0                    pypi_0    pypi
pywin32                   302             py310h2bbff1b_2    anaconda
pywinpty                  2.0.2           py310h5da7b33_0
pyyaml                    6.0                      pypi_0    pypi
pyzmq                     22.3.0          py310hd77b12b_2    anaconda
requests                  2.28.1          py310haa95532_0
retrying                  1.3.3              pyhd3eb1b0_2
scipy                     1.9.1                    pypi_0    pypi
seaborn                   0.11.2                   pypi_0    pypi
send2trash                1.8.0              pyhd3eb1b0_1
setuptools                61.2.0          py310haa95532_0    anaconda
six                       1.16.0             pyhd3eb1b0_1    anaconda
sniffio                   1.2.0           py310haa95532_1
snowflake-connector-python 2.7.12                   pypi_0    pypi
soupsieve                 2.3.1              pyhd3eb1b0_0
sqlite                    3.38.5               h2bbff1b_0    anaconda
stack_data                0.2.0              pyhd3eb1b0_0    anaconda
statsmodels               0.13.2                   pypi_0    pypi
tangled-up-in-unicode     0.2.0                    pypi_0    pypi
tenacity                  8.1.0                    pypi_0    pypi
terminado                 0.13.1          py310haa95532_0
testpath                  0.6.0           py310haa95532_0
tk                        8.6.12               h2bbff1b_0    anaconda
tornado                   6.1             py310h2bbff1b_0    anaconda
tqdm                      4.64.1                   pypi_0    pypi
traitlets                 5.1.1              pyhd3eb1b0_0    anaconda
typing-extensions         4.3.0           py310haa95532_0
typing_extensions         4.3.0           py310haa95532_0
tzdata                    2022a                hda174b7_0    anaconda
urllib3                   1.26.11         py310haa95532_0
vc                        14.2                 h21ff451_1    anaconda
visions                   0.7.5                    pypi_0    pypi
vs2015_runtime            14.27.29016          h5e58377_2    anaconda
wcwidth                   0.2.5              pyhd3eb1b0_0    anaconda
webencodings              0.5.1           py310haa95532_1
websocket-client          0.58.0          py310haa95532_4
werkzeug                  2.2.2              pyhd8ed1ab_0    conda-forge
wheel                     0.37.1             pyhd3eb1b0_0    anaconda
widgetsnbextension        4.0.3                    pypi_0    pypi
win_inet_pton             1.1.0           py310haa95532_0
wincertstore              0.2             py310haa95532_2    anaconda
winpty                    0.4.3                         4
xz                        5.2.5                h8cc25b3_1    anaconda
zipp                      3.8.1              pyhd8ed1ab_0    conda-forge
zlib                      1.2.12               h8cc25b3_2    anaconda
```

### OS

Windows 10

### Checklist

- [X] There is not yet another bug report for this issue in the [issue tracker](https://github.com/ydataai/pandas-profiling/issues)
- [X] The problem is reproducible from this bug report. [This guide](http://matthewrocklin.com/blog/work/2018/02/28/minimal-bug-reports) can help to craft a minimal bug report.
- [X] The issue has not been resolved by the entries listed under [Common Issues](https://pandas-profiling.ydata.ai/docs/master/pages/support_contrib/common_issues.html).

**Comment by fabclmnt** [comment]
Closed with pandas-profiling 3.6.2 version.

**Comment by EdwardMarsDS** [comment]
Just reran on 3.6.3 and it works great!  Thank you!
